### PR TITLE
Refactor/HDF-96 rewards git submodule

### PR DESCRIPTION
**Pull request by mystbrent**
## Description of Changes

Before this update, the repository explicitly includes halo-reward contracts to the source code to instantiate the Rewards contract. This can be improved by using git submodules to easily reference the halo-rewards contracts to its github repository counterpart.

Added changes also include:

[x] Fix nonce error that happens when running `npm start` on local (Done by giving different cron schedules to tasks)
[x] Add postinstall npm script to package.json to auto install halo-rewards submodule deps
[x] Make use of typescript path module aliases when importing halo-rewards repo to source code

With regards to deriving the signer account from only mnemonic option, I think it limits the option for the worker runner to only one option of using mnemonic. This update also includes resolving that issue by prioritizing the requirement of private key first before mnemonic. This way local tests can easily be done without having to do much effort.

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-96)

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
- [x] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally
- [x] I have formatted my code using format document in VSCode

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/tasks/UpdateAMMReward.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Contract } from '@ethersproject/contracts'
+import { Wallet } from '@ethersproject/wallet'
+import Rewards from '@halo-rewards/artifacts/contracts/Rewards.sol/Rewards.json'
+
+const rewardsContract = new Contract(
+	process.env.REWARDS_CONTRACT_ADDRESS,
+	Rewards.abi,
```
Comment (bitcoinbrisbane) [inline src/tasks/UpdateAMMReward.ts:7]:
```suggestion
	Rewards.abi
```

File: src/index.ts [hunk 2]
```diff
@@ -15,20 +15,24 @@ import cron from 'node-cron'
...
+cron.schedule(process.env.UPDATE_MINTER_REWARD_INTERVAL!, async () => UpdateMinterRewardTask(wallet))
 
-cron.schedule(process.env.RELEASE_REWARD_TASK_INTERVAL!, async () => ReleaseRewardTask(wallet))
+cron.schedule(process.env.UPDATE_AMM_REWARD_INTERVAL!, async () => UpdateAMMRewardTask(wallet))
 
 console.log(`Script is now running...`)
 console.table({
-  RELEASE_REWARD_TASK_INTERVAL: process.env.RELEASE_REWARD_TASK_INTERVAL,
+  UPDATE_MINTER_REWARD_INTERVAL: process.env.UPDATE_MINTER_REWARD_INTERVAL,
+  UPDATE_AMM_REWARD_INTERVAL: process.env.UPDATE_AMM_REWARD_INTERVAL,
```
Comment (bitcoinbrisbane) [inline src/index.ts:37]: Do we need this comma?
```suggestion
  UPDATE_AMM_REWARD_INTERVAL: process.env.UPDATE_AMM_REWARD_INTERVAL
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Some suggestions

File: src/tasks/UpdateAMMReward.ts (hunk 1 above)
Comment (mystbrent) [reply src/tasks/UpdateAMMReward.ts:7]: The dangling comma is not required but as far as I know I have been following this as part of JS best practices: https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

Please let me know you're thoughts on this :)

File: src/index.ts (hunk 2 above)
Comment (mystbrent) [reply src/index.ts:37]: The dangling comma is not required but as far as I know I have been following this as part of JS best practices: https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8

Please let me know you're thoughts on this :)

**Review by ctverceles — COMMENTED** [review]
@mystbrent This is working for me now locally but on kovan i get this:
image. Testing on kovan worked yesterday, I don't know if this is a Kovan network related thing. FYI @ernestpascual @schystz  if this looks familiar

**Comment by schystz** [comment]
> @mystbrent This is working for me now locally but on kovan i get this:
> image. Testing on kovan worked yesterday, I don't know if this is a Kovan network related thing. FYI @ernestpascual @schystz if this looks familiar

Yeah this looks like the same error when calling Rewards contract's write methods in Kovan.

**Comment by mystbrent** [comment]
Thank you for spotting this guys @ctverceles @schystz. I have tested in my local and I do not have the error.

**Comment by mystbrent** [comment]
I'll do a root cause analysis for this

**Comment by ctverceles** [comment]
> Thank you for spotting this guys @ctverceles @schystz. I have tested in my local and I do not have the error.
> 
> 

Yes local works fine for me as well, its kovan thats getting that weird err for me. I suspect its not cause of this repo but we have to find out why regardless

**Comment by ctverceles** [comment]
Ok kovan working now! https://kovan.etherscan.io/address/0xAC0e7b9C0A2ed0c4420D163F64a1C6eF48459d0f @mystbrent

**Review by ctverceles — APPROVED** [review]

**Pull request merged**